rocAL/source/readers/image/file_source_reader.cpp

```diff
@@ -157,27 +158,59 @@ Reader::Status FileSourceReader::subfolder_reading() {
     std::sort(entry_name_list.begin(), entry_name_list.end());
 
     auto ret = Reader::Status::OK;
-    for (unsigned dir_count = 0; dir_count < entry_name_list.size(); ++dir_count) {
-        std::string subfolder_path = _full_path + "/" + entry_name_list[dir_count];
-        filesys::path pathObj(subfolder_path);
-        if (filesys::exists(pathObj) && filesys::is_regular_file(pathObj)) {
-            // ignore files with non-image extensions
-            auto file_extension_idx = subfolder_path.find_last_of(".");
-            if (file_extension_idx != std::string::npos) {
-                std::string file_extension = subfolder_path.substr(file_extension_idx + 1);
-                std::transform(file_extension.begin(), file_extension.end(), file_extension.begin(),
-                               [](unsigned char c) { return std::tolower(c); });
-                if ((file_extension != "jpg") && (file_extension != "jpeg") && (file_extension != "png") && (file_extension != "ppm") && (file_extension != "bmp") && (file_extension != "pgm") && (file_extension != "tif") && (file_extension != "tiff") && (file_extension != "webp") && (file_extension != "wav"))
-                    continue;
+    if (!_file_list_path.empty()) {
+        std::ifstream fp(_file_list_path);
+        if (fp.is_open()) {
+            while (fp) {
+                std::string file_label_path;
+                std::getline(fp, file_label_path);
+                std::istringstream ss(file_label_path);
+                std::string file_path;
+                std::getline(ss, file_path, ' ');
+                file_path = _folder_path + "/" + file_path;
```

Comment:
What if the folder path is empty, and user passes the absolute path in the text file
Need to handle that case

Reply:
@shobana-mcw Should we add this case since its not used in RNNT training and might need more testing for edge cases?

Reply:
But since this is file source reader, in future also we might have this case, so better to handle

Reply:
We need to handle this.

Reply:
Added support for relative and absolute file paths in file list and tested it with both file list versions